libipl/p10/ipl0.C

```diff
@@ -153,6 +153,15 @@ static int update_hwas_state_callback(struct pdbg_target* target, void *priv)
 
 	} else if (ipl_type() == IPL_TYPE_NORMAL) {
 
+		if (strcmp(pdbg_target_class_name(target), "proc") == 0) {
```

Comment:
can;t we check if type == proc and ignore.

Reply:
The "pdbg_target_class_name" function returning the char* and l got instruction like "Don't go for C++ if really not required".

Reply:
I mean something like this, here we are doing the strcmp check for all proc and FC targets even though we have a type out there which we can use to avoid the comparison.

} else if (ipl_type() == IPL_TYPE_NORMAL && type == FRU_TYPE_PROC) {
    if (strcmp(pdbg_target_class_name(target), "proc") == 0) {
			if (ipl_is_master_proc(target)) {
				ipl_log(IPL_INFO,
					"Primary processor is guarded "
					"so, skipping to apply");
				return GUARD_PRIMARY_PROC_NOT_APPLIED;
			}
    }
    else
    {
		if (!set_or_clear_state(target, target_info->set_hwas_state)) {
			ipl_log(IPL_ERROR,
				"Failed to clear functional state of fru type 0x%x\n",
				type);
			//Unable to clear the functional state of HWAS attribute of the
			//target, so we need to stop
			return GUARD_TGT_NOT_FOUND;
		}    
    }
}
else if (ipl_type() == IPL_TYPE_NORMAL && type == FRU_TYPE_FC) && !small_core_enabled()) {
{
}

Reply:
> here we are doing the strcmp check for all proc

I am not getting your point, here we are not checking all proc. pdbg target is already found (LN 137) by using the guard record entity path and we are checking whether the identified target is a processor, and is the primary processor.

But, I will agree with you `strcmp` is costly than checking the TYPE attribute value.

Currently, we are applying all types of targets guard records (which are available in the GUARD partition) during normal ipl. If the guard record is FC target then, we need to mark the child of FC (i.e core) as non-functional if the system running in the small core mode.

and even, with your approach, we will have more `else if` blocks, right with duplicate code (set_or_clear_state)? no?

Reply:
agree more duplicate code but for me comparision is the issue. we have type already why not use it rather make a call to pdbg .  not worth to hold I guess will push unless someone disagrees.

Reply:
Yeah got your point on the comparison and addressed that in the latest push i.e using the TYPE attribute instead of the pdbg call to avoid string comparison.